### PR TITLE
Fix `updateSession` and header overwrite issues

### PR DESCRIPTION
**Pull request by tusharpandey13**
- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### 📋 Changes

This PR implements three complementary fixes to resolve updateSession reliability issues:

- **Fixed race condition in timestamp calculations** - Eliminated inconsistent `maxAge` values by using the same `updatedAt` timestamp throughout `calculateMaxAge()` instead of calling `this.epoch()` twice, preventing `Max-Age=0` cookies from timing inconsistencies
  - MODIFIED `src/server/session/abstract-session-store.ts`

- **Implemented conditional legacy cookie cleanup** - Added existence check before deleting legacy cookies to prevent unnecessary `Max-Age=0` cookies appearing in response headers during `updateSession()` calls
  - MODIFIED `src/server/session/stateless-session-store.ts`
  - MODIFIED `src/server/session/stateless-session-store.test.ts` (updated test mocks)

- **Fixed header overwrite in Pages Router** - Resolved session chunk loss by collecting all `set-cookie` headers into an array before setting them, preventing `res.setHeader()` from overwriting multiple cookies
  - MODIFIED `src/server/client.ts`
  - ADDED `src/server/updateSession-header-fix.test.ts` (comprehensive test suite)

### 📎 References

Fixes: #2316

### 🎯 Testing

**Automated:**
- Updated existing StatelessSessionStore unit tests to mock `getChunkedCookie()` properly for legacy cookie scenarios
- Added comprehensive test suite `updateSession-header-fix.test.ts` covering multiple set-cookie header scenarios, legacy cookie deletion, and edge cases
- All 308+ unit tests pass

**Manual:**
1. Create a Pages Router API route that calls `updateSession(req, res, session)` with chunked session data
2. Add `console.log(res.getHeaders())` immediately after the `updateSession()` call  
3. Before fix: Shows either `'set-cookie': ['appSession=; Path=/; Max-Age=0']` (unnecessary deletion) or missing session chunks (header overwrite)
4. After fix: Shows all session chunks properly preserved without unnecessary `Max-Age=0` deletion cookies

### CODE REVIEW

**Comment by tusharpandey13** [comment]
@claude please review this PR

File: src/server/updateSession-header-fix.test.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    // Mock the session store to simulate setting multiple cookies
+    vi.spyOn(client["sessionStore"], "set").mockImplementation(
+      async (_reqCookies, resCookies) => {
+        // Simulate StatelessSessionStore setting multiple cookies
+        resCookies.set("appSession", "updated_session_value");
+        resCookies.set("appSession.1", "chunk_data_here");
+
+        // If there's a legacy cookie to delete, simulate that too
+        if (Math.random() > 0.5) {
```
[the remark this answers is not in the document]
Comment (tusharpandey13) [reply src/server/updateSession-header-fix.test.ts]: done

File: src/server/client.ts [hunk 2]
```diff
@@ -929,7 +929,27 @@ export class Auth0Client {
...
         });
 
+        // Handle multiple set-cookie headers properly
+        // resHeaders.entries() yields each set-cookie header separately,
+        // but res.setHeader() overwrites previous values. We need to collect
+        // all set-cookie values and set them as an array.
+        const setCookieValues: string[] = [];
+        const otherHeaders: Record<string, string> = {};
+
         for (const [key, value] of resHeaders.entries()) {
```
[the remark this answers is not in the document]
Comment (tusharpandey13) [reply src/server/client.ts:941]: addressed, added a comment
